api/pkg/transformer/types/table/table.go

```diff
@@ -218,18 +218,60 @@ func (t *Table) Sort(sortRules []*spec.SortColumnRule) error {
 
 // UpdateColumnsRaw add or update existing column with values specified in columnValues map
 func (t *Table) UpdateColumnsRaw(columnValues map[string]interface{}) error {
-	df := t.DataFrame()
+	origColumns := t.Columns()
+	updateCol := map[string]bool{} //name of col that are updates
+	combinedColumns := make([]*series.Series, 0)
+
+	columnValues = broadcastScalar(columnValues, t.NRow())
+
+	//Check through all original columns for updates
+	for _, origColumn := range origColumns {
```

Comment:
nit: I think this part can be simplified a bit

```
	// Check through all original columns for updates
	for _, origColumn := range origColumns {
		origColumnName := origColumn.Series().Name
		val, update := columnValues[origColumnName]

		updatedColumn := origColumn
		if update {
			// Update original column
			colSeries, err := series.NewInferType(val, origColumnName)
			if err != nil {
				return err
			}
			updatedColumn = colSeries

			// Record down columns that are updates
			updateCol[origColumnName] = true
		}

		// Add column, either the origColumn or the updated one
		combinedColumns = append(combinedColumns, updatedColumn)
	}
```

In general, try to have less `else` statement.